File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        n = len(tickets)
        q = deque()
        for i in range(n):
            q.append(i)
        time = 0
        while True:
            if tickets[k] == 0:
                break
            curr = q[0]
            tickets[curr] -= 1
            time += 1
            q.popleft()
            if tickets[curr] > 0:
                q.append(curr)
        return time

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        int n = tickets.size();
        queue<int> q;
        for (int i = 0; i < n; i++) {
            q.push(i);
        }
        int time = 0;
        while (true) {
            if (tickets[k] == 0) {
                break;
            }
            int curr = q.front();
            tickets[curr]--;
            time++;
            q.pop();
            if (tickets[curr] > 0) {
                q.push(curr);
            }
        }
        return time;
    }
};